### PR TITLE
Only blobs on the Station Z-level count towards completion

### PR DESCRIPTION
**Pull request by ghost**
no more lavaland/any shuttle blobs

:cl: MoreRobustThanYou
balance: Only blobs on the Station Z-level count towards completion
/:cl:

i saw a blob on hippie today **_move it's core to lavaland_**. EUGHHH!

### CODE REVIEW

**Comment by Iatots** [comment]
Well how did it go?

**Comment by MoreRobustThanYou** [comment]
@latots a deathsquad was sent and the blob committed suicide by lava because the round was so long

File: code/game/gamemodes/blob/powers.dm [hunk 1]
```diff
@@ -16,24 +16,27 @@
...
 			if(M.client)
 				to_chat(src, "<span class='warning'>Someone could see your blob core from here!</span>")
-				return 0
+				return FALSE
 		var/turf/T = get_turf(src)
 		if(T.density)
 			to_chat(src, "<span class='warning'>This spot is too dense to place a blob core on!</span>")
-			return 0
+			return FALSE
+		if(T.z != ZLEVEL_STATION)
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/powers.dm]: how does this check fix anything? it's not like the blob eye can move on a shuttle...

Comment (MoreRobustThanYou) [reply code/game/gamemodes/blob/powers.dm]: you can still click on the shuttle tile

Comment (ChangelingRain) [reply code/game/gamemodes/blob/powers.dm]: which doesn't spawn your core there, so your ability to click the tile is irrelevant

**Comment by PKPenguin321** [comment]
If anything letting the blob spread out of the station should count doubly against the station but I guess this is a balance thing

**Comment by MoreRobustThanYou** [comment]
@PKPenguin321 yes but the blob's objective is to "take over the _**station**_"

File: code/game/gamemodes/blob/powers.dm (hunk 1 above)
Comment (ChangelingRain) [reply code/game/gamemodes/blob/powers.dm]: in any case the correct fix would be to keep the overmind from placing nodes on shuttles, which you haven't done

**Review by ChangelingRain — CHANGES_REQUESTED** [review]
figure out how they're doing it, then fix that instead of buckshotting all over the code in the hope something you did will fix the issue

File: code/game/gamemodes/blob/powers.dm [hunk 2]
```diff
@@ -131,6 +128,11 @@
 	set category = "Blob"
 	set name = "Create Node Blob (50)"
 	set desc = "Create a node, which will power nearby factory and resource blobs."
+	if(!T)
+		var/T = get_turf(src)
+	if(T.z != ZLEVEL_STATION)
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/powers.dm]: keep them from placing **_on shuttles_**

File: code/game/gamemodes/blob/powers.dm [hunk 3]
```diff
@@ -194,7 +198,7 @@
 		to_chat(src, "<span class='userdanger'>You have no core and are about to die! May you rest in peace.</span>")
 		return
 	var/area/A = get_area(T)
-	if(isspaceturf(T) || A && !A.blob_allowed)
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/powers.dm]: why did you remove that check

File: code/game/gamemodes/blob/theblob.dm [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
 /obj/structure/blob/Initialize()
 	var/area/Ablob = get_area(loc)
-	if(Ablob.blob_allowed) //Is this area allowed for winning as blob?
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/theblob.dm]: this should already cover all cases

File: code/game/gamemodes/blob/powers.dm [hunk 5]
```diff
@@ -198,7 +198,7 @@
 		to_chat(src, "<span class='userdanger'>You have no core and are about to die! May you rest in peace.</span>")
 		return
 	var/area/A = get_area(T)
-	if(isspaceturf(T) || istype(A, /area/shuttle))
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/powers.dm]: why did you remove the check you added

Comment (MoreRobustThanYou) [reply code/game/gamemodes/blob/powers.dm]: BECAUSE /area/shuttle HAS blob_allowed AS FALSE


REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE
